src/integrations/admin/helpscout-beacon.php

```diff
@@ -0,0 +1,328 @@
+<?php
+
+namespace Yoast\WP\SEO\Integrations\Admin;
+
+use WPSEO_Addon_Manager;
+use WPSEO_Admin_Asset_Manager;
+use WPSEO_Tracking_Server_Data;
+use WPSEO_Utils;
+use Yoast\WP\SEO\Conditionals\Admin_Conditional;
+use Yoast\WP\SEO\Conditionals\Admin_Page_Conditional;
+use Yoast\WP\SEO\Helpers\Options_Helper;
+use Yoast\WP\SEO\Integrations\Integration_Interface;
+
+/**
+ * Class WPSEO_HelpScout
+ */
+class HelpScout_Beacon implements Integration_Interface {
+
+	/**
+	 * The id for the beacon.
+	 *
+	 * @var string
+	 */
+	protected $beacon_id = '2496aba6-0292-489c-8f5d-1c0fba417c2f';
+
+	/**
+	 * The products the beacon is loaded for.
+	 *
+	 * @var array
+	 */
+	protected $products = [];
+
+	/**
+	 * Whether to asks the user's consent before loading in HelpScout.
+	 *
+	 * @var bool
+	 */
+	protected $ask_consent = true;
+
+	/**
+	 * The options helper.
+	 *
+	 * @var Options_Helper
+	 */
+	protected $options;
+
+	/**
+	 * The array of pages we need to show the beacon on with their respective beacon IDs.
+	 *
+	 * @var array
+	 */
+	protected $pages_ids;
+
+	/**
+	 * The array of pages we need to show the beacon on.
+	 *
+	 * @var array
+	 */
+	protected $base_pages = [
+		'wpseo_dashboard',
+		'wpseo_titles',
+		'wpseo_search_console',
+		'wpseo_social',
+		'wpseo_tools',
+		'wpseo_licenses',
+	];
+
+	/**
+	 * The current admin page
+	 *
+	 * @var string
+	 */
+	protected $page;
+
+	/**
+	 * Headless_Rest_Endpoints_Enabled_Conditional constructor.
+	 *
+	 * @param Options_Helper $options The options helper.
+	 */
+	public function __construct( Options_Helper $options ) {
+		$this->options     = $options;
+		$this->ask_consent = ! $this->options->get( 'tracking' );
+		$this->page        = filter_input( INPUT_GET, 'page', FILTER_SANITIZE_STRING );
+
+		foreach ( $this->base_pages as $page ) {
+			$this->pages_ids[ $page ] = $this->beacon_id;
+		}
+	}
+
+	/**
+	 * {@inheritDoc}
+	 */
+	public function register_hooks() {
+		add_action( 'admin_enqueue_scripts', [ $this, 'enqueue_help_scout_script' ] );
+		add_action( 'admin_footer', [ $this, 'output_beacon_js' ] );
+	}
+
+	/**
+	 * Enqueues the HelpScout script.
+	 */
+	public function enqueue_help_scout_script() {
+		// Make sure plugins can filter in their "stuff", before we check whether we're outputting a beacon.
+		$this->filter_settings();
+		if ( ! $this->is_beacon_page() ) {
+			return;
+		}
+
+		$asset_manager = new WPSEO_Admin_Asset_Manager();
```

Comment:
The asset manager can also be injected through the constructor. Saves one extra constructor call.